File: ai-server/app/models/drug.py
from sqlalchemy import Column, ForeignKey, Integer, String

from app.db.base import Base


class Drug(Base):
    __tablename__ = "drug"

    id = Column(Integer, primary_key=True, autoincrement=True)
    name = Column(String(100))
    e_name = Column(String(100))
    efficacy = Column(String(500))
    e_efficacy = Column(String(500))
    disease_const_id = Column(Integer, ForeignKey("disease_const.id"))

File: back/src/main/java/solutiona/challenge/pickaboo/domain/entity/Drug.java
package solutiona.challenge.pickaboo.domain.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.DynamicUpdate;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@DynamicUpdate
@Table(name = "drug")
public class Drug {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name")
    private String name;

    @Column(name = "e_name")
    private String eName;

    @Column(name = "efficacy")
    private String efficacy;

    @Column(name = "e_efficacy")
    private String eEfficacy;

    @JoinColumn(name = "disease_const_id")
    @ManyToOne(fetch = FetchType.LAZY)
    private DiseaseConst diseaseConst;

}
